api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +84,76 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getMostRecentQueue(org.openmrs.Patient)
+	 */
+	@Override
+	public PatientQueue getMostRecentQueue(Patient patient) {
+		return dao.getMostRecentQueue(patient);
+	}
+	
+	/**
+	 * This Method checks if a patient visit Number exists
+	 * 
+	 * @param visitNumber A visit Number in form of a string which will be check to determine if it
+	 *            exists
+	 * @return
+	 */
+	private boolean isVisitNumberIdExisting(Patient patient, String visitNumber) {
```

Comment:
Am wondering the use of this method if the generation algorithm is guaranteed to generate unique numbers.